pkg/engine/engine.go

```diff
@@ -492,10 +584,135 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 		}
 		atomic.AddUint64(&e.metrics.ChunksScanned, 1)
 	}
+
+	wgReverify.Wait()
 	wgDetect.Wait()
 	ctx.Logger().V(4).Info("finished scanning chunks")
 }
 
+// chunkSecretKey ties secrets to the specific detector that found them. This allows identifying identical
+// credentials extracted by multiple different detectors processing the same chunk. Or duplicates found
+// by the same detector in the chunk. Exact matches on lookup indicate a duplicate secret for a detector
+// in that chunk - which is expected and not malicious. Those intra-detector dupes are still verified.
+type chunkSecretKey struct {
+	secret       string
+	detectorInfo ahocorasick.DetectorInfo
+}
+
+func likelyDuplicate(ctx context.Context, val chunkSecretKey, dupes map[chunkSecretKey]struct{}) bool {
+	const similarityThreshold = 0.9
+
+	valStr := val.secret
```

Comment:
What happened to the exact match case? We can do a very efficient exact match lookup in the map without iteration.

Reply:
We are using a new key type:
```
// chunkSecretKey ties secrets to the specific detector that found them. This allows identifying identical
// credentials extracted by multiple different detectors processing the same chunk. Or duplicates found
// by the same detector in the chunk. Exact matches on lookup indicate a duplicate secret for a detector
// in that chunk - which is expected and not malicious. Those intra-detector dupes are still verified.
type chunkSecretKey struct {
	secret       string
	detectorInfo ahocorasick.DetectorInfo
}
```

So an exact match is actually okay given it means we found the same cred multiple times in the chunk with the same detector.

Reply:
@ahrav https://github.com/trufflesecurity/trufflehog/pull/2356/commits/96c6cd518d8cc8a28f77ce5e0caebc6692bb2b57 any reason the exact match case was removed in this commit? Should we add it back in?

Reply:
So the idea is if there is an identical match that means they DetectorKey AND secret are identical. If that's true then they are the same cred in the same detector. The same secret across multiple detectors would not be identical matches, but they would get caught by the Levenshtein similarity check.

Reply:
Probably worth to do something like 
```
		if valStr == dupe && val.detectorInfo != dupeKey.detectorInfo {
			ctx.Logger().V(2).Info(
				"found exact duplicate",
			)
			return true
		}
```

In the loop before hitting it with a similarity calculation?